resdk/scripts.py

```diff
@@ -209,52 +212,65 @@ def parse_annotation_file(annotation_file):
-                presample['descriptor']['geo']['organism'] = organism
-
-            experiment_type = EXPERIMENT_TYPE.get(annotations[fn]['SEQ_TYPE'].upper(), '')
-            if experiment_type:
-                presample['descriptor']['geo']['experiment_type'] = experiment_type
-
-            resolwe.api.presample(presample['id']).patch({'descriptor': presample['descriptor']})
-
-        else:
-            print("Error uploading {}".format(fn), file=sys.stderr)
+    for sample_n in annotations.keys():
+        input_ = {}
+        fw_reads = annotations[sample_n]['FASTQ_PATH'].split(',')
+
+        if set(fw_reads).issubset(set(all_new_read_files_uploaded)):
+            descriptor, descriptor_schema = None, None
+
+            if read_schema:
+                descriptor_schema = read_schema['slug']
+                descriptor = {
+                    'barcode': annotations[sample_n].get('BARCODE', None),
+                    'barcode_removed': True if annotations[sample_n].get('BARCODE_REMOVED', 'N').upper() == 'Y'
+                                       else False,
+                    'instrument_type': annotations[sample_n].get('INSTRUMENT', None),
+                    'seq_date': annotations[sample_n].get('SEQ_DATE', None),
+                }
+
+            # Paired-end reads
+            if annotations[sample_n]['PAIRED_END'] == 'Y' and annotations[sample_n]['FASTQ_PATH_PAIR']:
+                rw_reads = annotations[sample_n]['FASTQ_PATH_PAIR'].split(',')
+                slug = 'upload-fastq-paired'
+                input_['src1'] = fw_reads
+                input_['src2'] = [os.path.join(genialis_seq_dir, f) for f in rw_reads]
+                fn = input_['src1'] + input_['src2']
+
+            # Single-end reads
+            else:
+                slug = 'upload-fastq-single'
+                input_['src'] = fw_reads
+                fn = input_['src']
+
+            data = resolwe.run(slug,
+                               input=input_,
+                               descriptor=descriptor,
+                               descriptor_schema=descriptor_schema,
+                               data_name=sample_n)
+
+            if data:
+                for up_file in fn:
+                    uploaded_files.append(up_file)
+
+                presample = resolwe.api.presample.get(data=data.id)[0]
+
+                if 'geo' not in presample['descriptor']:
+                    presample['descriptor']['geo'] = {}
+
+                organism = ORGANISMS.get(annotations[sample_n]['ORGANISM'].upper(), '')
+                if organism:
+                    presample['descriptor']['geo']['organism'] = organism
+
+                experiment_type = EXPERIMENT_TYPE.get(annotations[sample_n]['SEQ_TYPE'].upper(), '')
+                if experiment_type:
+                    presample['descriptor']['geo']['experiment_type'] = experiment_type
+
+                resolwe.api.presample(presample['id']).patch({'descriptor': presample['descriptor']})
```

Comment:
This can be now written as `presample.update_descriptor(presample['descriptor'])`